### ISSUE TITLE
CardModel.is_editable() performs poorly with a large amount of data

### ISSUE THREAD

**Issue by AJ-HistoricEngland** [archesproject/arches#6530]
**Description**
The `CardModel.is_editable()` function determines if a card is editable (in the graph designer) by counting related tiles in the tiles table. Although the request uses an index scan, it has been measured to take 2-5 seconds to execute with 15+million tiles rows.

Where we 7-12 cards per graph, this can take 20-30 seconds to load the page - seems to occur on the graph page and search page.

**Possible Solutions**
1) May be able do a short lived cache to count tiles group by nodegroupid so there is a single hit and all subsequent is_editable calls during a page load can check the count list. Will need to look at the timings

2) new cardmodel field that is updated by the Card save/delete?

**Comment by chiatt** [comment]
@AJ-HistoricEngland, @azaroth42 wrote a fix for master #6320. It might be worth trying to cherry pick his commits into your branch and seeing how it improves performance in your instance.

**Comment by AJ-HistoricEngland** [comment]
@chiatt thanks for pointing that out. I'll have a play and see how much of a difference it makes 👍

**Comment by azaroth42** [comment]
For me it made my 2 *minute* page loads back down to a much more reasonable 2 seconds. (This was with 100s of millions of tiles)

**Comment by chiatt** [comment]
@AJ-HistoricEngland Have you had a chance to test this out? If so, is it okay to close this issue?

**Comment by AJ-HistoricEngland** [comment]
@chiatt I have implemented this and it is a little faster. However, it still takes too long as I have found another TileModel count earlier in the process. This again makes things run terribly.

Let's close this ticket as this specific issue is fixed. However, I think some kind of TileStats cache or table needs to be implemented to be able to give quick stats on the content of the Tile model to remove the need to do a .count() or .exists().
